odoo/addons/base/models/ir_model.py

```diff
@@ -470,10 +470,12 @@ def _related_field(self):
                 raise UserError(_("Unknown field name '%s' in related field '%s'") % (name, self.related))
             if index < last and not field.relational:
                 raise UserError(_("Non-relational field name '%s' in related field '%s'") % (name, self.related))
+            if self.store and not field.store and not field.compute:
```

Comment:
Are you sure that it's for intermediate fields only? What if `purchase_vendor_bill_id` is at the end of field list?

You also suggest to remove check for `self.store`. It should not be a problem for non-storable fields, no? I.e. field is not storable and hence no problem with recomputing the related fields in all records.

What about this 
```suggestion
            if self.store and not field._description_searchable:
```

Reply:
The last field in the list does not matter: you never have to search on that field.

Suppose your related field `r` is defined as `x.y.z`.  For the sake of simplicity, assume that both `x` and `y` are many2one fields.  But the reasoning is essentially the same for x2many fields.  There are three fields that may trigger the computation of the field:
* `x` is modified on `mods` → recompute `r` on `recs = mods`
* `y` is modified on `mods` → recompute `r` on `recs` that satisfy the domain `[('x', 'in', mods.ids)]`
* `z` is modified on `mods` → recompute `r` on `recs` that satisfy the domain `[('x.y', 'in', mods.ids)]`

As you can see, all the fields in the list *except the last one* must be searchable.  When possible, the ORM uses inverse fields instead of searching, but the inverse fields are nothing more than a cache for the search...